docs/pages/versions/unversioned/sdk/app-auth.md

```diff
@@ -21,57 +28,83 @@ For [managed](../../introduction/managed-vs-bare/#managed-workflow) apps, you'll
 
 Below is a set of example functions that demonstrate how to use `expo-app-auth` with the Google OAuth Sign-In provider.
 
+<SnackInline>
+
 ```js
-import { AsyncStorage } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import { AsyncStorage, Button, StyleSheet, Text, View } from 'react-native';
 import * as AppAuth from 'expo-app-auth';
 
-const config = {
+export default function App() {
+  let [authState, setAuthState] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      let cachedAuth = await getCachedAuthAsync();
+      if (cachedAuth && !authState) {
+        setAuthState(cachedAuth);
+      }
+    })();
+  }, []);
+
+  return (
+    <View style={styles.container}>
+      <Text>Expo AppAuth Example</Text>
+      <Text></Text>
+      <Button
+        title="Sign In with Google "
+        onPress={async () => {
+          let _authState = await signInAsync();
+          setAuthState(_authState);
+        }}
+      />
+      <Button
+        title="Sign Out "
+        onPress={async () => {
+          await signOutAsync(authState);
+          setAuthState(null);
+        }}
+      />
+      <Text>{JSON.stringify(authState)}</Text>
+    </View>
+  );
+}
+
+let styles = StyleSheet.create({
```

Comment:
```suggestion
const styles = StyleSheet.create({
```

Reply:
I think `const` is overused in most circumstances. 
https://overreacted.io/on-let-vs-const/